### PR TITLE
Trial: Postfix `as`

### PR DESCRIPTION
**Pull request by odersky**
Based on #10487

### CODE REVIEW

File: docs/docs/reference/contextual/derivation.md [hunk 1]
```diff
@@ -179,7 +179,7 @@ we need to implement a method `Eq.derived` on the companion object of `Eq` that
 a `Mirror[T]`. Here is a possible implementation,
 
 ```scala
-inline given derived[T](using m: Mirror.Of[T]) as Eq[T] = {
+inline given derived[T] => (m: Mirror.Of[T]) => Eq[T] = {
```
Comment (Jasper-M) [inline docs/docs/reference/contextual/derivation.md:182]: Missed one.

File: compiler/src/dotty/tools/dotc/core/TypeComparer.scala [hunk 2]
```diff
@@ -630,7 +630,7 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
             }
         }
         compareTypeLambda
-      case tp2 as OrType(tp21, tp22) =>
+      case tp2 @ OrType(tp21, tp22) =>
```
Comment (tgodzik) [inline compiler/src/dotty/tools/dotc/core/TypeComparer.scala:633]: Are we also reverting the possibility to use `as` instead of `@` ?

Comment (bishabosha) [reply compiler/src/dotty/tools/dotc/core/TypeComparer.scala:633]: no there are other pattern match bindings later on that still use `as`

Comment (tgodzik) [reply compiler/src/dotty/tools/dotc/core/TypeComparer.scala:633]: Thanks! I was super confused by this change.

**Pull request closed without merging**